### PR TITLE
Begin to Document Script Data

### PR DESCRIPTION
**Pull request by ravepossum**
This PR:
- Begins to document the structure used to store script data (currently named `ScriptManager`) and a lot of the related functionality.
- Adds a new constgen file for script commands, `scrcmd.json`
- Documents a couple script commands, `CheckSaveType` and `ScrollBG3`

Remaining items (in general, not necessarily for this PR):
- Documenting the rest of the `ScriptManager` struct (there's a few struct members that aren't documented). Both the struct itself, and the enumeration used in the accessor `ScriptManager_GetMemberPtr`.
- Creating preproc constants for the rest of the script ID offsets
- Documenting the remainder of the functions in `unk_0203E880.c`
- Renaming `unk_0203E880`

### CODE REVIEW

File: src/unk_0203E880.c
Comment (lhearachel) [inline src/unk_0203E880.c:102]: **polish:** Merge these into a conjunction.

File: src/unk_0203E880.c
Comment (lhearachel) [inline src/unk_0203E880.c:296]: **question:** Do we have constants for these bank IDs? What about the script NARC indices?

File: src/unk_0203E880.c
Comment (lhearachel) [inline src/unk_0203E880.c:303]: **suggestion:**

- `scriptsIndex` -> `scriptFile`
- `bankID` -> `textBank`

File: src/unk_0203E880.c
Comment (lhearachel) [inline src/unk_0203E880.c:555]: **suggestion:** Doesn't this routine check if a script refers to a single battle? In-context, at least, the call pattern seems to be to invoke this against a trainer's script.

File: src/unk_0203E880.c
Comment (lhearachel) [inline src/unk_0203E880.c:606]: **question:** How is this invoked? It seems like it operates on maps that... *aren't* the Iron Island / Floaroma Meadow maps?

File: src/unk_0203E880.c
Comment (lhearachel) [inline src/unk_0203E880.c:655]: **polish**: `const HiddenItem *item = gHiddenItems;`

File: src/unk_0203E880.c
Comment (lhearachel) [inline src/unk_0203E880.c:659]: **polish:** I assume that this does not match if you attempt to index against `gHiddenItems`?

File: src/scrcmd.c
Comment (ravepossum) [reply src/scrcmd.c:2445]: I might be missing something, but I don't think so. The way it works is that `directionX` is either 0 or 1 to indicate the direction the scroll happens in. I think 0 = right, and 1 = left. I could leave a comment to make that a little clearer.

File: src/unk_0203E880.c
Comment (ravepossum) [reply src/unk_0203E880.c:1]: I don't see why not. I was mostly holding off to get feedback on the `ScriptManager` name for the struct. But then again, it's not necessarily the whole focus of the file. Regardless, I'll toss out `script_manager.c`. `script.c` would work too.

File: src/unk_0203E880.c
Comment (ravepossum) [reply src/unk_0203E880.c:296]: I don't think we have constants for either of these yet. That's something I'm looking to do soon, fill out more of the scripting-related constants in general.

File: src/unk_0203E880.c
Comment (ravepossum) [reply src/unk_0203E880.c:555]: I can document the relevant script command that uses this soon, but it looks to me like it determines the trainer's "order" in a double battle based on what their trainer script ID is. Maybe this should be `Order` instead of `Index`?

Also returning a `BOOL` here is kind of weird - maybe it's worth changing it to return an `int`? It does match.

File: src/unk_0203E880.c
Comment (ravepossum) [reply src/unk_0203E880.c:659]: Unfortunately, yup, doesn't match.

File: src/unk_0203E880.c
Comment (ravepossum) [reply src/unk_0203E880.c:606]: It's called in the function that runs once a day to update day-based events. My guess is it's written that way so hidden items don't reappear on any maps that you're currently located in.

File: src/scrcmd.c
Comment (lhearachel) [reply src/scrcmd.c:2445]: Ah, I misread. Dyslexia moment.

File: src/unk_0203E880.c
Comment (lhearachel) [reply src/unk_0203E880.c]: I'd suggest that it returns an `int`, then, and probably the name `GetTrainerBattlerIndex` to implicitly tie it to the `BattleContext` fields.

**Review by cbt6 — COMMENTED** [review]
Prefer `GF_ASSERT(!x);` over `if (x) { GF_ASSERT(FALSE); }`

File: asm/macros/scrcmd.inc
Comment (cbt6) [inline asm/macros/scrcmd.inc:118]:
```suggestion
    .macro CallCommonScript scriptID
```

File: asm/macros/scrcmd.inc
Comment (cbt6) [inline asm/macros/scrcmd.inc:123]:
```suggestion
    .macro ReturnToMainScript
```

File: include/unk_0203E880.h
Comment (cbt6) [inline include/unk_0203E880.h:31]:
```suggestion
    SCRIPT_MANAGER_MAIN_CONTEXT,
    SCRIPT_MANAGER_SUB_CONTEXT,
```

File: include/unk_0203E880.h
Comment (cbt6) [inline include/unk_0203E880.h:77]:
```suggestion

enum {
    SCRIPT_CONTEXT_MAIN,
    SCRIPT_CONTEXT_SUB,
    NUM_SCRIPT_CONTEXTS
};
```

File: include/unk_0203E880.h
Comment (cbt6) [inline include/unk_0203E880.h:105]:
```suggestion
    u8 subCtxActive;
```

File: include/unk_0203E880.h
Comment (cbt6) [inline include/unk_0203E880.h:107]:
```suggestion
    u8 numActiveContexts;
```

File: include/unk_0203E880.h
Comment (cbt6) [inline include/unk_0203E880.h:122]:
```suggestion
    ApproachingTrainer trainers[2];
```

File: src/unk_0203E880.c
Comment (cbt6) [inline src/unk_0203E880.c:91]:
```suggestion
        scriptManager->ctx[SCRIPT_CONTEXT_MAIN] = ScriptContext_New(fieldSystem, scriptManager->scriptID);
```

File: src/unk_0203E880.c
Comment (cbt6) [inline src/unk_0203E880.c:98]:
```suggestion
        for (i = 0; i < NUM_SCRIPT_CONTEXTS; i++) {
```

File: src/unk_0203E880.c
Comment (cbt6) [inline src/unk_0203E880.c:172]:
```suggestion
ScriptContext * ScriptContext_CreateAndStart (FieldSystem * fieldSystem, u16 scriptID)
```

File: src/unk_0203E880.c
Comment (cbt6) [inline src/unk_0203E880.c:308]:
```suggestion
static void ScriptContext_LoadFromCurrentMap (FieldSystem * fieldSystem, ScriptContext * ctx)
```

File: src/unk_0203E880.c
Comment (cbt6) [inline src/unk_0203E880.c:464]:
```suggestion
static void * ScriptContext_LoadScripts (int headerID)
```

File: asm/macros/scrcmd.inc
Comment (ravepossum) [reply asm/macros/scrcmd.inc]: Went with `ReturnCommonScript` per discord conversation

**Comment by ravepossum** [comment]
Everything should be addressed + the files have been renamed now.

**Review by lhearachel — APPROVED** [review]
This looks mergeable to me; I'll let cbt give it another glean before merging.

**Comment by cbt6** [comment]
LGTM

**Pull request merged**
